Add Binding Group property to TileSet

98teg opened the pull request:
<!--
Pull requests should always be made for the `master` branch first, as that's
where development happens and the source of all future stable release branches.

Relevant fixes are cherry-picked for stable branches as needed.

Do not create a pull request for stable branches unless the change is already
available in the `master` branch and it cannot be easily cherry-picked.
Alternatively, if the change is only relevant for that branch (e.g. rendering
fixes for the 3.2 branch).
-->
This pull request is the result of the following proposal: [ #1260](https://github.com/godotengine/godot-proposals/issues/1260)

**Results:**

- **Editor**

![Captura de pantalla de 2020-07-29 13-30-03](https://user-images.githubusercontent.com/33127169/88795037-ac5cf580-d19f-11ea-9828-14ecd871fa54.png)
![Captura de pantalla de 2020-07-29 13-29-12](https://user-images.githubusercontent.com/33127169/88795076-b8e14e00-d19f-11ea-87de-c66ae1fb8158.png)
![Captura de pantalla de 2020-07-29 13-29-34](https://user-images.githubusercontent.com/33127169/88795085-baab1180-d19f-11ea-947b-4a32335d76e1.png)

- **Documentation**

![Captura de pantalla de 2020-07-29 13-33-06](https://user-images.githubusercontent.com/33127169/88795310-1c6b7b80-d1a0-11ea-9ed4-ae23d4f6b784.png)
![Captura de pantalla de 2020-07-29 13-33-30](https://user-images.githubusercontent.com/33127169/88795315-1ecdd580-d1a0-11ea-9e82-21e996985ccd.png)

98teg committed:
Add binding group property to TileSet

aaronfranke commented:
@98teg Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch. While there are no conflicts, rebasing is important so that reviewers can easily test the PR.

98teg commented:
@aaronfranke I have rebased, fixed an error that has emerged and tested the functionality. It still works, but I have encountered many crashes and unexpected behaviour while testing. I'm not sure if it is because TileSet is a bit unstable on master or if it is because of my changes. Please, let me know if anything needs to be fixed and thanks for considering this PR!

groud commented:
Sorry, I should have reviewed this earlier on.

The TileSet resource is having a full rewrite for 4.0. Autotiles will likely be removed in favor of terrains. Also, the linked proposal hasn't seen any feedback from other users, so I am not sure how often this behavior is needed. I mean, even on the provided screenshots, I believe that, most of the time, you will need transition tiles between the two terrains, which a terrain system handles better. So the proposal needs a little bit more discussion before a merge would be considered.

In any case, this cannot be merged into master, as the TileSet resources is completely different now in my WIP and the changes would be lost. We can still discuss the change for 3.x, but we need more feedback on the proposal.
